lib/assertions/tag-name.test.js

```diff
@@ -1,149 +1,44 @@
-        },
-        { tagName: "li" },
-        "p"
-    );
-});
-
-testHelper.assertionTests("refute", "tagName", function(
-    pass,
-    fail,
-    msg,
-    error
-) {
-    fail("for matching tag names", { tagName: "li" }, "li");
-    fail("for case-insensitive matching tag names", { tagName: "LI" }, "li");
-    fail("for case-insensitive matching tag names #2", { tagName: "LI" }, "li");
-    fail("for same casing matching tag names", { tagName: "li" }, "li");
-    pass("for non-matching tag names", { tagName: "li" }, "p");
-    pass("for substring matching tag names", { tagName: "li" }, "i");
-    pass("for case-insensitive non-matching tag names", { tagName: "li" }, "P");
-    pass(
-        "for case-insensitive substring mathcing tag names",
-        { tagName: "li" },
-        "i"
-    );
-
-    msg(
-        "fail with message",
-        "[refute.tagName] Expected tagName not to be li",
-        { tagName: "li" },
-        "li"
-    );
-
-    msg(
-        "fail with custom message",
-        "[refute.tagName] Oh well: Expected tagName not to be li",
-        { tagName: "li" },
-        "li",
-        "Oh well"
-    );
-
-    msg(
-        "fail if not passed arguments",
-        "[refute.tagName] Expected to receive at least 2 arguments"
-    );
-
-    msg(
-        "fail if not passed tag name",
-        "[refute.tagName] Expected to receive at least 2 arguments",
-        { tagName: "p" }
-    );
-
-    msg(
-        "fail if object does not have tagName property",
-        "[refute.tagName] Expected {  } to have tagName property",
-        {},
-        "li"
-    );
-
-    msg(
-        "fail with custom message if object does not have tagName property",
-        "[refute.tagName] Yes: Expected {  } to have tagName property",
-        {},
-        "li",
-        "Yes"
-    );
-
-    if (typeof document !== "undefined") {
-        pass("for DOM elements", document.createElement("li"), "p");
-    }
-
-    error(
-        "for matching tag names",
-        {
-            code: "ERR_ASSERTION",
-            operator: "refute.tagName"
-        },
-        { tagName: "li" },
-        "li"
-    );
+var assert = require("assert");
+var referee = require("../referee");
+
+describe("tagName", function() {
+    it("should pass for matching tag names", function() {
+        referee.assert.tagName({ tagName: "li" }, "li");
+    });
+
+    it("should pass for case-insensitive matching tag names", function() {
+        referee.assert.tagName({ tagName: "LI" }, "li");
+    });
+
+    it("should pass for case-insensitive matching tag names #2", function() {
+        referee.assert.tagName({ tagName: "li" }, "LI");
+    });
+
+    it("should pass for uppercase matching tag names", function() {
+        referee.assert.tagName({ tagName: "LI" }, "LI");
+    });
+
+    it("should fail for non-matching tag names", function() {
+        var actual;
+        try {
+            referee.assert.tagName({ tagName: "li" }, "p");
+        } catch (err) {
```

Comment:
You could use [assert.throws](https://nodejs.org/api/assert.html#assert_assert_throws_fn_error_message) here.